sunpy/map/mapbase.py

```diff
@@ -631,14 +631,18 @@ def observer_coordinate(self):
         """
         The Heliographic Stonyhurst Coordinate of the observer.
         """
+        missing_meta = {}
         for keys, kwargs in self._supported_observer_coordinates:
-            if all([k in self.meta for k in keys]):
+            meta_list = [k in self.meta for k in keys]
+            if all(meta_list):
                 return SkyCoord(obstime=self.date, **kwargs).heliographic_stonyhurst
+            elif any(meta_list) and not set(keys).isdisjoint(self.meta.keys()):
+                if type(kwargs['frame']) is not str:
```

Comment:
You should always try and use `isinstance` for type comparison:
```suggestion
                if not isinstance(kwargs['frame'], str):
```

Reply:
Done